stl/inc/format

```diff
@@ -701,8 +701,8 @@ public:
     public:
         template <class _Ty>
         explicit handle(_Ty& _Val) noexcept
-            : _Ptr(_STD addressof(_Val)),
-              _Format([](basic_format_parse_context<_CharType>& _Parse_ctx, _Context& _Format_ctx, const void* _Ptr) {
+            : _Ptr(_STD addressof(_Val)), _Format([](basic_format_parse_context<_CharType>& _Parse_ctx,
+                                                      _Context& _Format_ctx, const void* _Ptr) _STATIC_CALL_OPERATOR {

```

Comment:
There are x86 test failures here:

```
D:\build\out\inc\format(704,51): error: cannot initialize return object of type 'void (*)(basic_format_parse_context<_CharType> &, std::basic_format_context<std::back_insert_iterator<std::_Fmt_buffer<char>>, char> &, const void *)' (aka 'void (*)(basic_format_parse_context<char> &, std::basic_format_context<std::back_insert_iterator<std::_Fmt_buffer<char>>, char> &, const void *)') with an lvalue of type 'void (basic_format_parse_context<_CharType> &, std::basic_format_context<std::back_insert_iterator<std::_Fmt_buffer<char>>, char> &, const void *) __attribute__((thiscall))' (aka 'void (basic_format_parse_context<char> &, std::basic_format_context<std::back_insert_iterator<std::_Fmt_buffer<char>>, char> &, const void *) __attribute__((thiscall))')
  704 |             : _Ptr(_STD addressof(_Val)), _Format([](basic_format_parse_context<_CharType>& _Parse_ctx,
      |                                                   ^
D:\build\out\inc\format(2062,62): note: in instantiation of function template specialization 'std::basic_format_arg<std::basic_format_context<std::back_insert_iterator<std::_Fmt_buffer<char>>, char>>::handle::handle<const std::chrono::time_point<std::chrono::system_clock, std::chrono::duration<long long, std::ratio<1, 1000000>>>>' requested here
 2062 |             _Store_impl<_Erased_type>(_Arg_index, _Arg_type, static_cast<_Erased_type>(_Val));
```

`_Format` is a `__cdecl` function pointer, which is why this is x86-specific:

https://github.com/microsoft/STL/blob/202e3820d6390505855ce9a58a4bcee4f28feacd/stl/inc/format#L698

Presumably something about the `static` function call operator is interfering with the conversion from the lambda here. I suspect that Clang may not have considered the "omniconvertible" behavior (that MSVC originally implemented) when they added `static operator()`.

Reply:
This seems fixed in Clang 18 ([Godbolt link](https://godbolt.org/z/6zbPqo89d), despite that Clang-cl is not properly supported on Godbolt).

It seems that we should drop `_STATIC_CALL_OPERATOR` at this moment.